pkg/rbac/cache.go

```diff
@@ -6,22 +6,33 @@ import (
 
 	"github.com/driftprogramming/pgxpoolmock"
 	db "github.com/stolostron/search-v2-api/pkg/database"
-	v1 "k8s.io/client-go/kubernetes/typed/authentication/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+
+	authv1 "k8s.io/client-go/kubernetes/typed/authentication/v1"
 )
 
 // Cache used to optimize requests to the Kubernetes API server.
 type Cache struct {
-	authClient       v1.AuthenticationV1Interface // This allows tests to replace with mock client.
+	authClient       authv1.AuthenticationV1Interface // This allows tests to replace with mock client.
+	corev1Client     corev1.CoreV1Interface
+	kubeClient       kubernetes.Interface
+	resConfig        *rest.Config
 	tokenReviews     map[string]*tokenReviewCache
 	tokenReviewsLock sync.Mutex
+	userLock         sync.Mutex
 	shared           clusterScopedResources
+	users            map[string]*userData // UID:{userdata} UID comes from tokenreview
 	pool             pgxpoolmock.PgxPool
 }
```

Comment:
Some reorganization and minor changes. 
- Use a pointer `*` if possible for the clients to avoid duplicating the object.
- Renamed `userLock` to `usersLock` for consistency.
- Renamed `resConfig` to `restConfig`

```go
// Cache used to minimize requests to external APIs (Kubernetes and Database)
type Cache struct {
	tokenReviews     map[string]*tokenReviewCache
	tokenReviewsLock sync.Mutex
	shared           clusterScopedResources
	users            map[string]*userData // UID:{userdata} UID comes from tokenreview
	usersLock        sync.Mutex
	
	// Clients to external APIs.
	// Defining these here allow the tests to replace with a mock client.
	authClient       *authv1.AuthenticationV1Interface
	corev1Client     *corev1.CoreV1Interface
	kubeClient       *kubernetes.Interface
	pool             *pgxpoolmock.PgxPool  // Database client
	restConfig       *rest.Config
}
```